store/commitment/store.go

```diff
@@ -42,26 +44,28 @@ type CommitStore struct {
 }
 
 // NewCommitStore creates a new CommitStore instance.
-func NewCommitStore(multiTrees map[string]Tree, db store.RawDB, pruneOpts *store.PruneOptions, logger log.Logger) (*CommitStore, error) {
+func NewCommitStore(trees map[string]Tree, db store.RawDB, pruneOpts *store.PruneOptions, logger log.Logger) (*CommitStore, error) {
 	if pruneOpts == nil {
 		pruneOpts = store.DefaultPruneOptions()
 	}
 
 	return &CommitStore{
 		logger:       logger,
 		db:           db,
-		multiTrees:   multiTrees,
+		multiTrees:   trees,
 		pruneOptions: pruneOpts,
 	}, nil
 }
 
-func (c *CommitStore) WriteBatch(cs *store.Changeset) error {
-	for storeKey, pairs := range cs.Pairs {
-		tree, ok := c.multiTrees[storeKey]
+func (c *CommitStore) WriteBatch(cs *corestore.Changeset) error {
+	for storeKey, pairs := range cs.Changes {
+
+		key := internal.UnsafeBytesToStr(pairs.Actor)
+		tree, ok := c.multiTrees[key] //todo: see if we can avoid this hashing each time
 		if !ok {
 			return fmt.Errorf("store key %s not found in multiTrees", storeKey)
 		}
-		for _, kv := range pairs {
+		for _, kv := range pairs.StateChanges {
 			if kv.Value == nil {
```

Comment:
we should check if kv.Remov instead of doing kv.Value==nil (although it's correct in both ways, the former is more explicit)